addons/mail/static/src/components/file_uploader/file_uploader_tests.js

```diff
@@ -87,6 +88,49 @@ QUnit.test('no conflicts between file uploaders', async function (assert) {
     );
 });
 
+QUnit.test('handles pre-upload removals of attachments', async function (assert) {
+    assert.expect(1);
+
+    const attachmentUploadedPromise = makeTestPromise();
+    await this.start({
+        async mockFetch(resource, init) {
+            const res = this._super(...arguments);
+            if (resource === '/web/binary/upload_attachment') {
+                await attachmentUploadedPromise;
+            }
+            return res;
+        }
+    });
+    const fileUploader = await this.createFileUploaderComponent();
+    const file1 = await createFile({
+        name: 'text1.txt',
+        content: 'hello, world',
+        contentType: 'text/plain',
+    });
+    const file2 = await createFile({
+        name: 'text2.txt',
+        content: 'hello, world',
+        contentType: 'text/plain',
+    });
+    inputFiles(
+        fileUploader.el.querySelector('.o_FileUploader_input'),
+        [file1, file2]
+    );
+    const attachmentModel = this.env.models['mail.attachment'];
+    await nextAnimationFrame();
+    const attachment = attachmentModel.find(attachment =>
+        attachment.isTemporary &&
+        attachment.filename === 'text2.txt'
+    );
+    await attachment.remove();
+    attachmentUploadedPromise.resolve();
+    assert.strictEqual(
+        attachmentModel.all().length,
+        1,
+        'Uploaded file should be the only attachment created'
+    );
```

Comment:
Similar remark, would it be possible to see the only 1 remaining attachment on the interface?

Reply:
Question: what part of this test crashes without the fix?

Reply:
An error is thrown from the file uploader when the promise is resolved.